conda/core/repodata.py

```diff
@@ -413,7 +413,7 @@ def fetch_repodata(url, schannel, priority,
         mtime = getmtime(cache_path)
     except (IOError, OSError):
         log.debug("No local cache found for %s at %s", url, cache_path)
-        if use_cache:
+        if context.offline or use_cache:
```

Comment:
Does the test you wrote actually fail if we don't make this modification?  If so, then I think there has to be something wrong with the logic below around line 431.